tests/integration_tests/test_cli.py

```diff
@@ -75,6 +76,52 @@ def test_that_oom_kills_are_reported():
         )
 
 
+@pytest.mark.usefixtures("copy_poly_case")
+def test_disk_sync_wait():
+    with open("poly_eval.py", "w", encoding="utf-8") as f:
+        f.write(
+            dedent(
+                """\
+                #!/usr/bin/env python
+                import json
+                import subprocess
+                from pathlib import Path
+
+                def _load_coeffs(filename):
+                    with open(filename, encoding="utf-8") as f:
+                        return json.load(f)["COEFFS"]
+
+                def _evaluate(coeffs, x):
+                    return coeffs["a"] * x**2 + coeffs["b"] * x + coeffs["c"]
+
+                def run_process():
+                    Path("run_me").write_text(
+                        "import time\\n"
+                        "from pathlib import Path\\n"
+                        "if __name__ == '__main__':\\n"
+                        "\\ttime.sleep(0.5)\\n"
```

Comment:
I think we should really avoid testing this with sleeps as this creates tests that are unreliable and runs long. Instead, we can inject removing or changing the file as separate steps, either as forward models (`MOVE_FILE` or `DELETE_FILE`) or custom hooked workflows.